tools/scons/site_scons/site_tools/emscripten/emscripten.py

```diff
@@ -31,21 +39,25 @@ def generate(env, emscripten_path=None, **kw):
               'EMCC_FORCE_STDLIBS', 'EMCC_ONLY_FORCED_STDLIBS', 'EM_PORTS', 'IDL_CHECKS', 'IDL_VERBOSE']:
     if os.environ.get(var):
       env['ENV'][var] = os.environ.get(var)
+
   try:
     emscPath = emscripten_path.abspath
   except:
     emscPath = emscripten_path
 
-  env.Replace(CC=os.path.join(emscPath, "emcc"))
-  env.Replace(CXX=os.path.join(emscPath, "em++"))
```

Comment:
Actualy, on further investigation can we simplify this a lot by just removing `emscripten_path` and `emscPath` completely, and assuming emcc is in the PATH?   Seems like a reasonable assumption to me.

Then we can just do `env.Replace(CC='emcc')`?  Much simpler.

If we want we could have emscons inject add the emscripten root to the PATH but that also seems maybe unnecessary?

Reply:
I agree that it is bad for the tool to use an internal function like I did, but I would like for it to be able to provide all the environment variables that the other commands like `emmake` and `emconfigure` do, particularly for `pkg-config` to work.

So would it be better to duplicate all the variable definitions here? I was thinking that could be a burden if a new variable was added to `tools/building.py`

Reply:
I could also use `emscons` to provide the values, and just expand the list of environment variables to apply from the parent process (line 31) to include them 🤔

I think it is not very good either because SCons is not supposed to use `CC` or other similar environment variables automatically, and in isolation that is what the tool would be doing. But that would only be relevant if the tool was not being used with `emscons`, which seems unlikely, so maybe it's ok (?)